playbooks/technical-planning.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Technology choices
+description: How to write technical plans at Artsy
+---
+
+# Technical planning
+
+Technical planning is a crucial part of our engineering process. Writing tech plans helps us:
+- Keep track of technical decisions that we've made, and the associated rationale.
+- Streamline a team's development of a feature and align within and across teams.
+- Communicate and collaborate effectively on technical decisions in an async manner.
+- Think through problems in a way that allows developers to practice technical decision-making & grow.
+- Get feedback early on in the development process, to avoid costly U-turns further down the line.
+
+Developers should use their judgment when deciding whether or not to write a tech plan, but we recommend them when planning:
+1. Technical design for medium to large features
+1. Architectural changes or new design patterns
+1. Changes affecting other teams' products or engineers' workflows
+1. New integrations or services
+
+If the feature you're working on follows our best-practices and doesn't require technical decisions larger than what would fit in a single PR, it's OK to skip.
+
+If you're working on a small feature or it doesn't fit the criteria above but you still want to write one, that's fine! Feel free to keep it small.
```

Comment:
👍 I would stress that it is always better to document a decision (even a small one), especially if it might be not obvious for people who will work on this in future. Short [Y-statements](https://medium.com/olzzio/y-statements-10eb07b5a177) can be enough for this purpose.

Reply:
Hadn't heard of Y-statements or ([architecture haikus](https://www.georgefairbanks.com/blog/comparch-wicsa-2011-panel-discussion-and-haiku-tutorial/) 😄) before. Cool!

Reply:
Maybe a bit off-topic, but I've always liked our own [engineering playbook's approach](https://github.com/artsy/README/blob/main/playbooks/engineer-workflow.md#pull-requests) to this, which encourages a similar (but lightweight version) of this for PRs.

Reply:
I can leave a note here about the importance of documenting decisions (at least somewhere!).